math/fibonacci.c

```diff
@@ -1,23 +1,70 @@
-#include <stdio.h>
+/**
+ * @file
+ * @brief Program to print the nth term of the Fibonacci series.
+ * @details
+ * Fibonacci series generally starts from 0 and 1. Every next term in
+ * the series is equal to the sum of the two preceding terms.
+ */
 
-// Fibonnacci function
-int fib(int number)
+#include <assert.h>  /// for assert()
+#include <stdio.h>   /// for input, output
+#include <stdlib.h>  /// for exit() - to exit the program
+
+/**
+ * @brief Determines the nth Fibonacci term
+ * @param number - n in "nth term" and it can't be negative as well as zero
+ * @return nth term in unsigned type
+ * @warning
+ * Only till 47th and 48th fibonacci element can be stored in 'int' and
+ * 'unsigned' respectively (takes more than 20 seconds to print)
+ */
+unsigned int fib(int number)
 {
-    if (number == 1 || number == 2)
+    // Check for negative integers
+    if (number <= 0)
+    {
+        fprintf(stderr, "Illegal Argument Is Passed!\n");
+        exit(EXIT_FAILURE);
+    }
+
+    // Base conditions
+    if (number == 1)
+        return 0;
+
+    if (number == 2)
         return 1;
-    else
-        return fib(number - 1) + fib(number - 2);
+
+    // Recursive call to the function
+    return fib(number - 1) + fib(number - 2);
+}
+
+/**
+ * @brief self-test implementation
+ * @return void
+ */
+static void test()
+{
+    assert(fib(5));
+    assert(fib(2));
+    assert(fib(9));
 }
 
+/**
+ * @brief Main function
+ * @return 0 on exit
+ */
 int main()
 {
     int number;
 
-    // Asks for the number that is in n position in Fibonnacci sequence
-    printf("Number: ");
-    scanf("%d", &number);
+    // Performing the test
+    test();
+    printf("Tests passed...\n");
 
-    printf("%d \n", fib(number));
+    // Getting n
+    printf("Enter n to find nth fibonacci element: ");
+    scanf("%d", &number);
```

Comment:
What if the user inputted something other than a valid number? It's quite interesting to write input checks, and you should certainly consider adding some, but it's probably a bit too much, just ask me if you need assistance.😄 [And here's a great resource](https://sekrit.de/webdocs/c/beginners-guide-away-from-scanf.html)

Reply:
This is very much helpful!

Reply:
If characters other than numbers are entered, they are detected by the first base condition and the code exits.

Reply:
> If characters other than numbers are entered, they are detected by the first base condition and the code exits.

That would be the case if you had initialized `number` to, say, `-1` and `scanf()` didn't parse anything. But no, since `number` is not initialized except by this `scanf()`, bad things can happen. I want challenge you to write a proper input check, it would be a great exercise.